File: server/src/main/java/com/cloudmeta/storage/service/FolderService.java
package com.cloudmeta.storage.service;

import java.util.List;
import java.util.UUID;

import com.cloudmeta.storage.dto.folder.BreadcrumbItem;
import com.cloudmeta.storage.dto.folder.CreateFolderRequest;
import com.cloudmeta.storage.dto.folder.FolderResponse;
import com.cloudmeta.storage.dto.folder.UpdateFolderRequest;

public interface FolderService {

    FolderResponse createFolder(CreateFolderRequest request, String userEmail);

    List<FolderResponse> getFolders(UUID parentId, String userEmail);

    FolderResponse getFolderById(UUID folderId, String userEmail);

    FolderResponse renameFolder(UUID folderId, UpdateFolderRequest request, String userEmail);

    void deleteFolder(UUID folderId, String userEmail);

    List<BreadcrumbItem> getBreadcrumbs(UUID folderId, String userEmail);
}

File: client/src/services/folderService.js
import api from "./api";

export const getFolders = async (parentId = null) => {
  const params = parentId ? { parentId } : {};
  const response = await api.get("/folders", { params });
  return response.data;
};

export const getFolderById = async (folderId) => {
  const response = await api.get(`/folders/${folderId}`);
  return response.data;
};

export const createFolder = async ({ name, parentId }) => {
  const response = await api.post("/folders", { name, parentId: parentId || null });
  return response.data;
};

export const renameFolder = async (folderId, { name }) => {
  const response = await api.put(`/folders/${folderId}`, { name });
  return response.data;
};

export const deleteFolder = async (folderId) => {
  const response = await api.delete(`/folders/${folderId}`);
  return response.data;
};

export const getBreadcrumbs = async (folderId) => {
  const response = await api.get(`/folders/${folderId}/breadcrumbs`);
  return response.data;
};

